logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -574,4 +577,20 @@ abstract class UserSessionScopeCommon(
                 it.createDirectories(dataStoragePaths.assetStoragePath.value.toPath())
         }
     }
+
+    override val coroutineContext: CoroutineContext = SupervisorJob()
+
+    fun onInit() {
+        launch {
+            // TODO: Add a public start function to the Managers
+            incrementalSyncManager
+            slowSyncManager
+
+            callRepository.updateOpenCallsToClosedStatus()
```

Comment:
I'm thinking if this shouldn't be done by the `CallScope` or similar.

Or at least something a bit more "higher-level" than the repository. Because this sets precedent for other repositories being called here directly and I can imagine `UserSessionScopeCommon` doing far too much stuff.

But it works for now!

Reply:
Indeed! I thought on adding the UseCase (I did the implementation of it) but ended up just removing it and going straight with the repository for now 🤔